File: src/test/java/io/mfedirko/fixture/DynamoLessons.java
package io.mfedirko.fixture;

import io.mfedirko.common.infra.dynamodb.DynamoContactRequest;
import io.mfedirko.common.infra.dynamodb.DynamoLesson;
import lombok.experimental.UtilityClass;

import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.util.List;
import java.util.regex.Pattern;

@UtilityClass
public class DynamoLessons {
    public static final List<DynamoLesson> DATA;

    static {
        // convert csv with AI-generated mock data to list
        BufferedReader reader = new BufferedReader(new InputStreamReader(DynamoLessons.class.getResourceAsStream("/lesson.csv")));
        DATA = reader.lines()
                .skip(1)
                .map(s -> s.split(Pattern.quote("|")))
                .map(s -> DynamoLesson.builder()
                        .id(s[0].substring(0, 4))
                        .creationTimestampMillis(Long.parseLong(s[1]))
                        .author(s[2])
                        .title(s[3])
                        .category(s[4])
                        .description(s[5])
                        .build())
                .toList();
    }
}

File: src/test/kotlin/io/mfedirko/fixture/DynamoLessons.kt
package io.mfedirko.fixture

import io.mfedirko.common.infra.dynamodb.DynamoLesson
import java.io.BufferedReader
import java.io.InputStreamReader
import java.util.regex.Pattern

object DynamoLessons {
    var DATA: List<DynamoLesson>? = null

    init {
        // convert csv with AI-generated mock data to list
        val reader = BufferedReader(InputStreamReader(DynamoLessons::class.java.getResourceAsStream("/lesson.csv")!!))
        DATA = reader.lines()
            .skip(1)
            .map { s: String ->
                s.split(Pattern.quote("|").toRegex()).dropLastWhile { it.isEmpty() }
                    .toTypedArray()
            }
            .map { s: Array<String> ->
                DynamoLesson().apply {
                    id = s[0].substring(0, 4)
                    creationTimestampMillis = s[1].toLong()
                    author = s[2]
                    title = s[3]
                    category = s[4]
                    description = s[5]
                }
            }
            .toList()
    }
}